compiler/damlc/daml-doc/src/DA/Daml/Doc/Extract.hs

```diff
@@ -39,13 +47,35 @@ import qualified Data.Text                                 as T
 import           Data.Tuple.Extra                          (second)
 import Data.Either
 
--- | Parse, and process documentation in, a dependency graph of modules.
-mkDocs
-    :: IdeOptions
+-- | Options that affect doc extraction.
+data ExtractOptions = ExtractOptions
+    { eo_qualifyTypes :: QualifyTypes
+        -- ^ qualify non-local types
+    , eo_simplifyQualifiedTypes :: Bool
```

Comment:
I'm not convinced this option is a good idea in a Haskell-based language since you cannot refer to `X.Y.z` as `Y.z` when you're in `X`. (In OCaml this would be a great option though.)

Reply:
I see this option as a viable compromise to achieve a short but qualified name . Use case being something like `import X.Y.Z.Blah as Blah` - imported `stuff` may come out as `Z.Blah.stuff` but that is better than the full name or simply `stuff`.

Reply:
I disagree that `Z.Blah.stuff` is better than `X.Y.Z.Blah.stuff` or `stuff`. If there's actually something whose fully qualified names is `Z.Blah.stuff`, this can be incredibly confusing.

Reply:
The links are still pointing to the right place though, it's just the text that is a little more qualified. This really is for the use case where fully unqualified names (the default) are too ambiguous (e.g. with OCaml convention), but fully qualified names are overly verbose.